src/main/scala/io/moia/scalaHttpClient/HttpClient.scala

```diff
@@ -123,7 +123,8 @@ abstract class HttpLayer[LoggingContext](
     case response @ HttpResponse(StatusCodes.Success(_), _, _, _)                => Future.successful(HttpClientSuccess(response))
     case response @ HttpResponse(StatusCodes.BadRequest, _, HttpEntity.Empty, _) => Future.successful(HttpClientError(response))
     case response @ HttpResponse(StatusCodes.BadRequest, _, _, _)                => Future.successful(DomainError(response))
-    case response @ HttpResponse(code, _, _, _) if shouldBeRetried(code)         => retryWithConfig(tryNumber, httpRequest, response, deadline)
+    case response @ HttpResponse(StatusCodes.ClientError(_), _, _, _)            => retryWithConfig(tryNumber, httpRequest, response, deadline)
```

Comment:
I don't think that we should retry client errors by default :thinking:

Ahhh nevermind seems like the default retry count for client errors is set to 0, right?